pkg/sql/colexec/agg/stddevpop/stddevpop.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stddevpop
+
+import (
+	"math"
+	"reflect"
+	"strconv"
+	"unsafe"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
+
+func String(b []byte) (s string) {
+	pbytes := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	pstring := (*reflect.StringHeader)(unsafe.Pointer(&s))
+	pstring.Data = pbytes.Data
+	pstring.Len = pbytes.Len
+	return
+}
+
+func (stp *Stddevpop[T1, T2]) TransFloat64(a any) float64 {
```

Comment:
You should not type switch on T.   In this case, do not mix decimal with other types.    Type switch has a run time cost.

Reply:
Yes, I will teach him a way of writing with the generics.